Makefile

```diff
@@ -232,24 +232,37 @@ protogen_show: ## A simple `find` command that shows you the generated protobufs
 protogen_clean: ## Remove all the generated protobufs.
 	find . -name "*.pb.go" | grep -v -e "prototype" -e "vendor" | xargs -r rm
 
+PROTOC = protoc --experimental_allow_proto3_optional --go_opt=paths=source_relative
+PROTOC_SHARED = $(PROTOC) -I=./shared
+
```

Comment:
Not actionable: Nice but I can't wait to see `buf` being used here. 
A couple of config files and here we would essentially have a oneliner to generate all the structs from protos

Reply:
```
# IMPROVE: Look into using buf in the future; https://github.com/bufbuild/buf.
```